libs/wallet/src/web3-react/updater.ts

```diff
@@ -61,44 +66,83 @@ function _useWalletDetails(account?: string, standaloneMode?: boolean): WalletDe
 }
 
+
 function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
   const { provider } = useWeb3React()
   const { account, chainId } = walletInfo
   const [safeInfo, setSafeInfo] = useState<GnosisSafeInfo>()
-  const { isReadOnly } = useSafeAppsSdkInfo() || {}
+  const safeAppsSdk = useSafeAppsSdk()
 
   useEffect(() => {
-    const update = () => {
+    const updateSafeInfo = async () => {
       if (chainId && account && provider) {
-        getSafeInfo(chainId, account, provider)
-          .then((_safeInfo) =>
-            setSafeInfo({
+        if (safeAppsSdk) {
+          const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+          setSafeInfo((prevSafeInfo) => {
+            const { isReadOnly } = appsSdkSafeInfo
+            if (!prevSafeInfo) {
+              // We don't have prevSafeInfo for counterfactual safes as the SDK can't find a deployment
+              // of this safe on-chain
+              const { safeAddress, threshold, owners } = appsSdkSafeInfo
+              return {
+                chainId,
+                address: safeAddress,
+                threshold,
+                owners,
+                isReadOnly,
+              }
+            }
+
+            return {
+              ...prevSafeInfo,
               isReadOnly,
-              ..._safeInfo,
-            })
-          )
-          .catch(() => {
-            console.debug(`[WalletUpdater] Address ${account} is likely not a Safe (API didn't return Safe info)`)
-            setSafeInfo(undefined)
+            }
           })
+        } else {
+          getSafeInfo(chainId, account, provider)
+            .then((_safeInfo) => {
+              const { address, threshold, owners } = _safeInfo
+              setSafeInfo((prevSafeInfo) => ({
+                ...prevSafeInfo,
+                chainId,
+                address,
+                threshold,
+                owners,
+                isReadOnly: false,
+              }))
+
+              // Being here means that we don't run in an iframe over Safe's AppCommunicator,
+              // so we need to slow down the interval to not spam the RPC
+              clearInterval(shortSafeInfoInterval);
+              longSafeInfoInterval = setInterval(updateSafeInfo, SAFE_INFO_LONG_INTERVAL);
+            })
+            .catch(() => {
+              console.debug(`[WalletUpdater] Address ${account} is likely not a Safe (API didn't return Safe info)`)
+                setSafeInfo(undefined)
+            })
+        }
       } else {
         setSafeInfo(undefined)
       }
     }
 
-    const interval = setInterval(update, SAFE_INFO_UPDATE_INTERVAL)
+    shortSafeInfoInterval = setInterval(updateSafeInfo, SAFE_INFO_SHORT_INTERVAL)
 
-    update()
+    updateSafeInfo()
 
-    return () => clearInterval(interval)
-  }, [setSafeInfo, chainId, account, provider, isReadOnly])
+    return () => {
+      clearInterval(shortSafeInfoInterval)
+      clearInterval(longSafeInfoInterval);
```

Comment:
should you nullify it so it doesn't happen that the promise above resolves after the hook is re-rendered?